File: PrimaMateria/PrimaMateria/Classes/XTRCornerImageView.m
//
//  XTRCornerImageView.m
//  PrimaMateria
//
//  Created by Jerry Porter on 1/5/2011.
//  Copyright (c) 2016 xTrensa. All rights reserved.
//

@implementation XTRCornerImageView

- (void)createLayer {
    self.gradientLayer = [[CAGradientLayer alloc] init];
    self.gradientLayer.bounds = self.bounds;
    self.gradientLayer.position = CGPointMake(self.bounds.size.width / 2, self.bounds.size.height / 2);
    [self.layer insertSublayer: self.gradientLayer atIndex: 0];
    self.layer.cornerRadius = 10.0f;
    self.layer.masksToBounds = YES;
}

- (void)awakeFromNib {
    [super awakeFromNib];
    [self createLayer];
}

#pragma mark - Memory Management Methods


@end

File: PrimaMateria/PrimaMateria/Classes/XTRCornerImageView.swift
//
//  XTRCornerImageView.swift
//  PrimaMateria
//
//  Created by Jerry Porter on 4/14/16.
//  Copyright © 2016 xTrensa. All rights reserved.
//

@objc class XTRCornerImageView : UIImageView {
    private var gradientLayer:  CAGradientLayer!
    
    // MARK: - Initialization Methods
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)!
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.createLayer()
    }
    
    // MARK: - Internal Methods
    
    func createLayer() {
        self.gradientLayer = CAGradientLayer.init()
        self.gradientLayer.bounds = self.bounds
        self.gradientLayer.position = CGPointMake(self.bounds.size.width / 2, self.bounds.size.height / 2)
        self.layer.insertSublayer(self.gradientLayer, atIndex: 0)
        self.layer.cornerRadius = 10.0
        self.layer.masksToBounds = true
    }
}
